src/server/object_services/object_server.js

```diff
@@ -1928,11 +1928,13 @@ async function _complete_object_multiparts(obj, multipart_req) {
         }
         md5.update(Buffer.from(mp.md5_b64, 'base64'));
         const mp_parts = parts_by_mp[mp._id.toHexString()];
-        _complete_next_parts(mp_parts, context);
-        used_multiparts.push(mp);
-        for (const part of mp_parts) {
-            used_parts.push(part);
+        if (mp_parts) { // handles case where multipart is of size 0 - so mp_parts is undefined
```

Comment:
no error should be thrown here? AWS handles this situation the same?

Reply:
@romayalon Yes, same scenario works in AWS. So we shouldn't fail as well.